rest/api.go

```diff
@@ -285,6 +285,10 @@ func (h *handler) handleProfiling() error {
 		if isCPUProfile {
 			base.Infof(base.KeyAll, "... ending CPU profile")
 			pprof.StopCPUProfile()
+			if fileCloseError := h.server.GetCpuPprofFile().Close(); fileCloseError != nil {
+				base.Warnf("Error closing profile file: %v", fileCloseError)
+			}
+			h.server.SetCpuPprofFile(nil)
```

Comment:
I still think there's a race here between start and stop, due to the lock being released between lines 288 and 291.

We'll need an operation to both close the existing file, and nil the file descriptor whilst holding the lock.

Reply:
Closing the open file descriptor and pointing it to a nil reference can be implemented in CloseCpuPprofFile with acquiring and releasing the lock. Changes are available in latest revision. Warning the error whilst closing file descriptor is also included in this method.